Update Quirk urls for twitter.com to x.com

stwrt opened the pull request:
#### bdf58da41cdd0f25547387247ca564d8abb2d49e
<pre>
Update Quirk urls for twitter.com to x.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=304111">https://bugs.webkit.org/show_bug.cgi?id=304111</a>
<a href="https://rdar.apple.com/problem/166444405">rdar://problem/166444405</a>

Reviewed by Tim Nguyen.

twitter.com appears to redirect to x.com now. Updating quirks urls.

* Source/WebCore/page/Quirks.cpp:
(WebCore::handleTwitterXQuirks):

Canonical link: <a href="https://commits.webkit.org/304508@main">https://commits.webkit.org/304508@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dea8abeaf0e0efd44c625fb5d483c93cf86feed7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135488 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7866 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46781 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143182 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/87171 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/bf1fb29c-7138-4bac-9a17-0fb8c0719930) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137357 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8503 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7713 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103535 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/70889 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/61023b51-be49-4b10-8aa2-372ce3dc9813) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138434 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6101 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121447 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84404 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/686bf458-94e4-49c3-9d55-037351271d84) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5878 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3488 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3792 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115081 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39633 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145932 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7551 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40204 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111901 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7590 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6335 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112273 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28549 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5735 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117749 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/61451 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7603 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35865 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7350 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71149 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7569 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7451 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
Can we check if we still need them before changing the URL?

karlcow reviewed (approved):
This is fixing only the comment. 
The last time it has been checked when the effective domain names have been changed. 
https://github.com/WebKit/WebKit/pull/37216/changes

It would be great to check again at regular pace if quirks are needed, but that's kind of true for any quirks. 

Thanks for fixing the comments which had been forgotten it seems.

stwrt committed:
Update Quirk urls for twitter.com to x.com
https://bugs.webkit.org/show_bug.cgi?id=304111
rdar://problem/166444405

Reviewed by Tim Nguyen.

twitter.com appears to redirect to x.com now. Updating quirks urls.

* Source/WebCore/page/Quirks.cpp:
(WebCore::handleTwitterXQuirks):

Canonical link: https://commits.webkit.org/304508@main
